src/modules/transaction/core/use-cases/withdraw.use-case.ts

```
@@ -0,0 +1,32 @@
+import { SupportedFiatCurrency } from 'src/utils/enums';
+import { OrderInput } from '../../api/graphql/input-types/order-input';
+import { Order } from '../../api/graphql/object-types/order-response';
+import { InvalidCurrencyError } from '../errors/invalid-currency.error';
+import { DRWithdrawUseCase } from './dr-withdraw.use-case';
+import { MXWithdrawUseCase } from './mx-withdraw.use-case';
+
+export interface WithdrawUseCaseDto extends OrderInput {
+  userId: string;
+}
+
+export class WithdrawUseCase {
+  constructor(
+    private readonly drWithdrawUseCase: DRWithdrawUseCase,
+    private readonly mxWithdrawUseCase: MXWithdrawUseCase
+  ) {}
+
+  public async execute(input: WithdrawUseCaseDto): Promise<Order> {
+    const { from_currency, to_currency } = input;
+
+    if (!Object.values(SupportedFiatCurrency).includes(from_currency as SupportedFiatCurrency))
+      throw new InvalidCurrencyError();
+    if (!Object.values(SupportedFiatCurrency).includes(to_currency as SupportedFiatCurrency))
+      throw new InvalidCurrencyError();
+
+    if (from_currency === SupportedFiatCurrency.MXN || to_currency === SupportedFiatCurrency.MXN)
+      return await this.mxWithdrawUseCase.execute(input);
+
+    if (from_currency === SupportedFiatCurrency.DOP || to_currency === SupportedFiatCurrency.DOP)
+      return await this.drWithdrawUseCase.execute(input);
+  }
```

You should throw an error if the end of the method is reached because this means that none of the previous cases has been matched